src/Core/Domain/Discount/Command/UpdateDiscountCommand.php

```diff
@@ -0,0 +1,366 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Discount\Command;
+
+use DateTimeImmutable;
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\CurrencyId;
+use PrestaShop\PrestaShop\Core\Domain\Customer\ValueObject\CustomerId;
+use PrestaShop\PrestaShop\Core\Domain\Discount\Exception\DiscountConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Discount\ValueObject\DiscountId;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CombinationConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationIdInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\NoCombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Money;
+
+class UpdateDiscountCommand
+{
+    private ?array $localizedNames = null;
+    private ?int $priority = null;
+    private ?bool $active = null;
+    private ?DateTimeImmutable $validFrom = null;
+    private ?DateTimeImmutable $validTo = null;
+    private ?int $totalQuantity = null;
+    private ?int $quantityPerUser = null;
+    private ?string $description = null;
+    private ?string $code = null;
+    private ?CustomerId $customerId = null;
+    private ?bool $highlightInCart = null;
+    private ?bool $allowPartialUse = null;
+    private ?DecimalNumber $percentDiscount = null;
+    private ?Money $amountDiscount = null;
+    private ?ProductId $productId = null;
+    private ?CombinationIdInterface $combinationId = null;
+    private ?int $reductionProduct = null;
+    private DiscountId $discountId;
+
+    public function __construct(int $discountId)
+    {
+        $this->discountId = new DiscountId($discountId);
+    }
+
+    /**
+     * @param array<int, string> $localizedNames
+     */
+    public function setLocalizedNames(?array $localizedNames): self
+    {
+        if (null === $localizedNames) {
+            $this->localizedNames = null;
+        }
+        foreach ($localizedNames as $languageId => $name) {
+            $this->localizedNames[(new LanguageId($languageId))->getValue()] = $name;
+        }
```

Comment:
```suggestion
        $this->localizedNames = $localizedNames;
```
Couldn't you simplify this? You create ValueObjects to get their value instantly so in the end the array is still based on integer values. Is it to ensure the indexes used are correct via the `LanguageId`, if so I understand the purpose but it's probably overkill IMHO 🤔